dashboard/admin.py

```diff
@@ -94,9 +103,11 @@ def _courseviewoption(self, obj):
     def course_link(self, obj):
         return format_html('<a href="{}">Link</a>', obj.absolute_url)
 
-    def change_view(self, request, object_id, form_url='', extra_content=None):
-        self.readonly_fields = self.readonly_fields + ('canvas_id',)
-        return super(CourseAdmin, self).change_view(request, object_id)
+    def get_readonly_fields(self, request: HttpRequest, obj: Any | None = ...) -> tuple[str, ...]:
+        readonly_fields: tuple[str, ...] = self.readonly_fields
+        if obj is not None:
+            readonly_fields += ('canvas_id',)
+        return readonly_fields
```

Comment:
This logics work. basically the `obj` contains `canvas_id`. can we make check if the obj contains 'canvas_id` or value of the canvas_id rather than just checking for None?

Reply:
I'll verify, but I think `obj` is the model instance. I think it's `None` if we're creating a new object.

Reply:
Verified. `obj` is `None` on create, `Course` when modifying.

Reply:
Ok .